spec/support/integrity_checker_helper.rb

```diff
@@ -0,0 +1,8 @@
+# frozen_string_literal: true
+
+module IntegrityCheckerHelper
+  def problem_message(attribute, expected, actual, plural = false)
+    negative_do = plural ? "don't" : "doesn't"
+    "#{attribute} #{negative_do} match, expected: #{expected.inspect}, actual: #{actual.inspect}"
+  end
```

Comment:
This looks good - though I would move this function into the test rather than create a specific helper  file for it (this'll keep it nearer to what it tests).

I also think we have only one instance of a plural (organisations) so I'd suggest just testing that manually rather than having this helper cope with that.

I think the thing we risk a bit is that we're writing the same logic in our test as in our application code and never test that. Since the logic is pretty trivial it doesn't seem a big concern though.

Reply:
I've moved it into the test file.